Spigot-Server-Patches/0604-Implemented-BlockFailedDispenseEvent.patch

```diff
@@ -0,0 +1,46 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: TheViperShow <29604693+TheViperShow@users.noreply.github.com>
+Date: Wed, 22 Apr 2020 09:40:38 +0200
+Subject: [PATCH] Implemented BlockFailedDispenseEvent
+
+
+diff --git a/src/main/java/net/minecraft/server/BlockDispenser.java b/src/main/java/net/minecraft/server/BlockDispenser.java
+index c239265581972d67c7567d7545d62499ebadd195..08f16fe32c12cb22ee5d793ac40b83c9398faf74 100644
+--- a/src/main/java/net/minecraft/server/BlockDispenser.java
++++ b/src/main/java/net/minecraft/server/BlockDispenser.java
+@@ -48,7 +48,14 @@ public class BlockDispenser extends BlockTileEntity {
+         int i = tileentitydispenser.h();
+ 
+         if (i < 0) {
+-            worldserver.triggerEffect(1001, blockposition, 0);
++            // Paper start - BlockFailedDispenseEvent is called here
++            org.bukkit.block.Block block = worldserver.getWorld().getBlockAt(blockposition.getX(), blockposition.getY(), blockposition.getZ());
```

Comment:
Would personally move that to a protected method that returns if it should play the effect.

protected triggerBlockFailedDispenseEvent(BlockPosition blockPosition)..

Reply:
yeah right, this should go into CraftEventFactory, will do that later